### ISSUE TITLE
Must the optional chain end with `.?propName`?

### ISSUE THREAD

**Issue by brad4d** [nicolo-ribaudo/proposal-optional-chaining-assignment#1]
Would this be legal syntax?

```js
maybeUndef?.method().prop = something;
```

**Comment by nicolo-ribaudo** [comment]
Yes, this proposal re-uses the existing optional chaining syntax with no additional restrictions other than "it cannot end with a function call". This is so that developers can re-using their existing knowledge about assignment to member expressions and about optional chaining, without having to remember that when used together there are arbitrary additional restricitons.
